gradle/plugins/src/main/kotlin/com.hedera.gradlebuild.patch-modules.gradle.kts

```diff
@@ -0,0 +1,154 @@
+extraJavaModuleInfo {
+    failOnAutomaticModules = true // Only allow Jars with 'module-info' on all module paths
+
+    module("com.esaulpaugh:headlong", "headlong")
+    module("com.github.spotbugs:spotbugs-annotations", "com.github.spotbugs.annotations")
+    module("com.google.errorprone:error_prone_annotations", "com.google.errorprone.annotations")
+    module("com.google.guava:failureaccess", "com.google.common.util.concurrent.internal")
+    module("com.google.guava:guava", "com.google.common") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.logging")
+    }
+    module("com.google.protobuf:protobuf-java", "com.google.protobuf") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.logging")
+        requires("jdk.unsupported")
+    }
+    module("com.google.protobuf:protobuf-javalite", "com.google.protobuf") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.logging")
+        requires("jdk.unsupported")
+    }
+    module("io.grpc:grpc-netty-shaded", "grpc.netty.shaded") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.logging")
+        requires("jdk.unsupported")
+        ignoreServiceProvider("reactor.blockhound.integration.BlockHoundIntegration")
+    }
+    module("io.grpc:grpc-protobuf-lite", "grpc.protobuf.lite")
+    module("io.grpc:grpc-protobuf", "grpc.protobuf")
+    module("io.grpc:grpc-stub", "io.grpc.stub") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.logging")
+    }
+    module("io.perfmark:perfmark-api", "io.perfmark")
+    module("javax.annotation:javax.annotation-api", "java.annotation") {
+        exportAllPackages()
+        mergeJar("com.google.code.findbugs:jsr305")
+    }
+    module("org.jetbrains:annotations", "org.jetbrains.annotations")
+
+    // Full protobuf only
+    module("com.google.api.grpc:proto-google-common-protos", "com.google.api.grpc.common")
+
+    // Testing only
+    module("com.fasterxml.jackson.core:jackson-annotations", "com.fasterxml.jackson.annotations")
+    module("com.fasterxml.jackson.core:jackson-core", "com.fasterxml.jackson.core")
+    module("com.fasterxml.jackson.core:jackson-databind", "com.fasterxml.jackson.databind") {
+        exportAllPackages()
+        requireAllDefinedDependencies()
+        requires("java.sql")
+    }
+    module("io.github.cdimascio:java-dotenv", "java.dotenv")
+    module("io.github.json-snapshot:json-snapshot", "json.snapshot")
+    module("junit:junit", "junit")
+    module("org.mockito:mockito-core", "org.mockito")
+    module("org.mockito:mockito-junit-jupiter", "org.mockito.junit.jupiter")
+    module("org.objenesis:objenesis", "org.objenesis")
+}
+
+dependencies {
+    javaModulesMergeJars("com.google.code.findbugs:jsr305:3.0.2")
+}
```

Comment:
Can you please explain purpose of this, when we have above
```
module("javax.annotation:javax.annotation-api", "java.annotation") {
        exportAllPackages()
        mergeJar("com.google.code.findbugs:jsr305")
}
```

Reply:
This is needed to define a version for `jsr305`. The above only contains IDs pointing at Jars. Before that configuration can be used, Jars need to be fetched by their version. Usually all versions are ind `sdk/build.gradle.kts` now. But this one does not "really exist" because it is merged. So it does not have a "Module Name" to address it.

But this is super complicated 😞 and we should get rid of this. It's all only here because both Jars contain the `javax.annotation` package. Ideally we could check the code if we can change something about which annotations are used to get rid of this.

I'll put this into a follow up issue.